pkg/build/controller/build/build_controller.go

```diff
@@ -1528,6 +1551,162 @@ func (bc *BuildController) createBuildRegistryConfigMapSpec(build *buildv1.Build
 	return cm
 }
 
+func (bc *BuildController) controllerConfigWorker() {
+	for {
+		if quit := bc.controllerConfigWork(); quit {
+			return
+		}
+	}
+}
+
+func (bc *BuildController) controllerConfigWork() bool {
+	key, quit := bc.controllerConfigQueue.Get()
+	if quit {
+		return true
+	}
+	defer bc.controllerConfigQueue.Done(key)
+
+	err := bc.handleControllerConfig()
+	bc.handleControllerConfigError(err, key)
+	return false
+}
+
+// handleControllerConfig synchronizes the build controller config in the cluster
+// `build.config.openshift.io` instance and any referenced data in the `openshift-config` namespace.
+func (bc *BuildController) handleControllerConfig() error {
+	// Make sure that we have synced the controller config store
+	// This will always be false if the controller is not running on an openshift/installer cluster,
+	// and therefore does not have the service cert signing controller.
+	//
+	// TODO: Remove this once all tests run on openshift/installer clusters
+	if !waitForCacheSync(10*time.Second, bc.controllerConfigStoreSynced) {
+		glog.V(2).Info("builds.config.openshift.io not synced")
+		glog.V(4).Info("disabling service signing CA injection for builds")
+		bc.injectServiceCA = false
+		return nil
+	}
+	bc.injectServiceCA = true
+	config, err := bc.configLister.Get("cluster")
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if config == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	if len(config.Spec.AdditionalTrustedCA.Name) == 0 {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	additionalCA, err := bc.configMapStore.ConfigMaps("openshift-config").Get(config.Spec.AdditionalTrustedCA.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if additionalCA == nil {
+		bc.additionalTrustedCAData = nil
+		return nil
+	}
+	bc.additionalTrustedCAData = additionalCA.Data
+	return nil
+}
+
+func (bc *BuildController) handleControllerConfigError(err error, key interface{}) {
+	if err != nil {
+		glog.V(2).Infof("Failed to update controller config: %v", err)
+	}
+	bc.controllerConfigQueue.Forget(key)
+}
+
+func (bc *BuildController) controllerConfigAdded(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigUpdated(old, cur interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigDeleted(obj interface{}) {
+	bc.controllerConfigChanged()
+}
+
+func (bc *BuildController) controllerConfigChanged() {
+	bc.controllerConfigQueue.Add("openshift-config")
+}
+
+func (bc *BuildController) configMapAdded(obj interface{}) {
+	configMap, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("object was not a configMap: %+v", obj))
+		return
+	}
+	if configMap == nil {
+		return
+	}
+	config, err := bc.configLister.Get("cluster")
```

Comment:
I've decided to use the BuildConfig object from the lister - since this is a cache there shouldn't be a big performance hit, and it ensures the data we use in these checks are updated in line with the cache.